### PR TITLE
Fixes to SOCmapper

### PR DESCRIPTION
**Pull request by lizgzil**
---

# Description

Addressing #81

- Add output of soc name if desired (defaults to not output it - so no breaking changes)
- Update some READMEs accordingly
- more job title cleaning (I'm sure we could do something much more sophisticated..)
- test for job title cleaning

# Instructions for Reviewer

In order to test the code in this PR you need to ...

Please pay special attention to ...

# Checklist:

- [ ] I have refactored my code out from `notebooks/`
- [ ] I have checked the code runs
- [ ] I have tested the code
- [ ] I have run `pre-commit` and addressed any issues not automatically fixed
- [ ] I have merged any new changes from `dev`
- [ ] I have documented the code
  - [ ] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [ ] I have explained this PR above
- [ ] I have requested a code review

### CODE REVIEW

**Comment by lizgzil** [comment]
Some OJO job titles to the SOC 2020 6 digit they map to - look ok!
ger - Fashion', 'Marketing managers'), ('Sales Manager', 'Sales managers'), ('Financial Accountant', 'Financial accountants '), ('Marketing Manager B2C eCommerce', 'Business development managers'), ('Graduate Sustainability/Energy Engineering role', 'Civil engineers n.e.c.'), ('Environmental, Health and Safety Assistant', 'Occupational health managers and officers'), ('Management Accountant ', 'Bookkeepers, payroll managers and wage clerks n.e.c.'), ('Digital Marketing Executive', 'Marketing consultants'), ('Fermentation Technician            ', 'Biological laboratory technicians'), ('Graduate Sales and Operations Coordinator', 'Sales administrators'), ('HGV Class 2 Driver - CAT C', 'Large goods vehicle drivers'), ('Weighbridge Administrator - Contract', 'Weighers, graders and sorters '), ('Graduate Sustainability/Energy Engineering role', 'Civil engineers n.e.c.'), ('Packaging Development Manager South', 'Managers in storage and warehousing'), ('Packaging Developer', 'Performance make-up artists'), ('Maintenance Manager', 'Metal working production and maintenance fitters n.e.c.'), ('Engineering Team Leader', 'Skilled metal, electrical and electronic trades supervisors'), ('Senior Sustainability Consultant', 'Sustainability officers'), ('Senior Building Surveyor', 'Building control surveyors'), ('Disability Employment Programme Manager', 'Welfare and housing associate professionals n.e.c.'), ('Sustainability Consultant', 'Sustainability officers'), ('Sustainability Consultant', 'Sustainability officers'), ('Senior Sustainability Consultant', 'Sustainability officers'), ('Digital Marketing Manager', 'Marketing managers'), ('Senior Sustainability Consultant', 'Sustainability officers'), ('Transport Engineer (HGV)', 'Heavy and commercial vehicle technicians'), ('Continuous Improvement Specialist- Manufacturing', 'Industrial and production engineers'), ('Health and Safety Officer, Atherstone', 'Health and safety managers and officers n.e.c.'), ('CHC Senior Community Practitioner', 'Social workers n.e.c.'), ('Senior Estates Advisor Capital Investment', 'Stockbrokers'), ('waste operations manager', 'Waste disposal and environmental services managers n.e.c.'), ('Natural Capital Woodland Creation Officer', 'Forest and woodland officers'), ('Business Analyst', 'Business analysts'), ('Marketing Manager  &#163;48-60K  Sustainability Tech Start-Up ', 'Marketing managers'), ('Business Development Manager', 'Business development managers'), ('Chartered Building Surveyor', 'Chartered surveyors n.e.c.'), ('Environmental Manager', 'Environmental professionals n.e.c.'), ('Senior Consultant', 'Social workers n.e.c.'), ('DevOps / System Administrator', 'Network and systems administrators'), ('CHIEF OFFICER OPERATIONS AND TRADED SERVICES', 'Financial managers and directors n.e.c.'), ('IT Engineering Manager', 'Computer system and equipment installers and servicers'), ('Lettings Administrator', 'Records clerks and assistants n.e.c.'), ('Computer Vision Engineer', 'Photographers, audio-visual and broadcasting equipment operators n.e.c.'), ('Building Surveyor', 'Building control surveyors'), ('Acoustic Consultant - North London', 'Acoustic engineers '), ('Junior DevOps Engineer  - Bristol or Remote', 'DevOps engineers'), ('Accounts Officer', 'Bookkeepers, payroll managers and wage clerks n.e.c.'), ('Sustainability Consultant', 'Sustainability officers'), ('Sustainability and Planning Consultant', 'Sustainability officers'), ('Contact Centre Team Leader', 'Customer service supervisors'), ('Nursery Manager required', 'Early education and childcare services managers'), ('Recruitment Consultant - Health &amp; Safety - London - 30-40k', 'Recruitment consultant '), ('Building Surveyor', 'Building control surveyors'), ('Simulation Engineer ', 'Avionics technicians '), ('EU Sustainability Project Engineering Manager', 'Civil engineers n.e.c.'), ('Senior .Net Developer/Architect', 'Software developers')]

File: dap_prinz_green_jobs/pipeline/green_measures/occupations/README.md [hunk 1]
```diff
@@ -21,10 +21,16 @@ from dap_prinz_green_jobs.pipeline.green_measures.occupations.soc_map import SOC
...
-soc_mapper.get_soc(job_titles=["data scientist", "Assistant nurse", "Senior financial consultant - London"])
->>> [('2425', 'data scientist'), ('6141', 'assistant nurse'), ('3534', 'financial consultant')]
+job_titles=["data scientist", "Assistant nurse", "Senior financial consultant - London"]
+
+soc_mapper.get_soc(job_titles, return_soc_name=True)
+>>> [((('2433/02', 'Data scientists'), ('2433', 'Actuaries, economists and statisticians'), '2425'), 'data scientist'), ((('6131/99', 'Nursing auxiliaries and assistants n.e.c.'), ('6131', 'Nursing auxiliaries and assistants'), '6141'), 'assistant nurse'), ((('2422/02', 'Financial advisors and planners'), ('2422', 'Finance and investment analysts and advisers'), '3534'), 'financial consultant')]
 ```
 
+The output for one job title is in the format `(((SOC 2020 Extension code, SOC 2020 Extension name), (SOC 2020 4-digit code, SOC 2020 4-digit name), SOC 2010 code), cleaned job title)`.
+
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/occupations/README.md:31]: this is really helpful!

File: dap_prinz_green_jobs/pipeline/green_measures/occupations/occupations_data_processing.py [hunk 2]
```diff
@@ -27,3 +29,119 @@ def process_green_timeshare_soc(green_timeshares: pd.DataFrame()) -> pd.DataFram
...
+    """
+    Will apply a bunch of cleaning to a job title
+    - removing certain things (locations or work type after a "-")
+    - fixes some unicode &#163; -> £
+    - Removes text after "£""
+
+    Assumption: weird bad stuff comes after dashes or £ signs.
+    So this won't work well for e.g "£30k Data Scientist" or "Remote - London Data Scientist"
+
+    This isn't perfect, but should hopefully help quite a few examples
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/occupations/occupations_data_processing.py:93]: fair enough but looks like it deals with a ton of messy job titles!

File: dap_prinz_green_jobs/pipeline/green_measures/occupations/soc_map.py [hunk 3]
```diff
@@ -326,6 +343,9 @@ def find_most_likely_soc(
         2. Get the 4-digit SOCs of the good (>top_n_sim_threshold) matches in the top n most similar.
         3. If there are a few of these (>=minimum_n) and over a certain proportion (>minimum_prop) of these are the same at the 4 digit level - use this as the SOC.
                 This will return (soc, the job titles given for this same soc)
+
+        Returns data in the format ((soc_2020_6, soc_2020_4, soc_2010), job_title) or None
+        If pathway 1. (above) isn't true then the output will be ((None, soc_2020_4, None), job_title) and job_title will be a set of multiple
```
Comment (india-kerle) [inline dap_prinz_green_jobs/pipeline/green_measures/occupations/soc_map.py:348]: as you're returning multiple soc name options is there one we should pick downstream for analysis? I just assume the first one?

File: dap_prinz_green_jobs/tests/test_occupation_measures.py [hunk 4]
```diff
@@ -49,3 +52,16 @@ def test_occupation_measures():
         )
         == 3
     )
+
+
+def test_job_title_cleaner():
```
Comment (india-kerle) [inline dap_prinz_green_jobs/tests/test_occupation_measures.py:57]: this test passed for me

**Review by india-kerle — APPROVED** [review]
Looks great! Especially like the description of the outputs + I think the cleaning makes sense.

File: dap_prinz_green_jobs/pipeline/green_measures/occupations/soc_map.py (hunk 3 above)
Comment (lizgzil) [reply dap_prinz_green_jobs/pipeline/green_measures/occupations/soc_map.py:348]: yes I think so

**Comment by lizgzil** [comment]
Final commit outputs a dict for the SOC 2020 code to names - handy when linking back

**Pull request merged**